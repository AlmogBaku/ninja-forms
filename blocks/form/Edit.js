import React from "react";
import { SelectControl, Placeholder } from "@wordpress/components";
import { InspectorControls } from "@wordpress/block-editor";
import useForms from "./useForms";
import FormPreviewIFrame from "./FormPreviewIFrame";
import { NinjaIcon } from "./icon";
import PropTypes from "prop-types";

const BlockPlaceholder = props => {
	return (
		<Placeholder
			icon={<div style={{ marginRight: "10px" }}>{NinjaIcon}</div>}
			label="Ninja Form"
			instructions="Display a form"
			style={{ alignItems: "center" }}
		>
			<div
				style={{
					margin: "auto",
					width: "50%"
				}}
			>
				{props.children}
			</div>
		</Placeholder>
	);
};

BlockPlaceholder.propTypes = {
	children: PropTypes.element.isRequired
};

/**
 * Form chooser select control
 */
export const ChooseForm = ({ formId, forms, onChange, labelText }) => {
	const { asSelectOptions } = useForms({ forms });
	const options = React.useMemo(() => {
		let opts = asSelectOptions();
		return [...[{ label: "---" }], ...opts];
	}, [forms, asSelectOptions]);
	return (
		<SelectControl
			label={labelText ? labelText : "Select Form"}
			value={formId}
			onChange={onChange}
			options={options}
		/>
	);
};

ChooseForm.propTypes = {
	formId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	forms: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
	onChange: PropTypes.func.isRequired,
	labelText: PropTypes.string
};

/**
 * Ninja Forms block UI
 */
export default function Edit({
	formId,
	forms,
	labelText,
	setAttributes,
	siteUrl,
	previewToken
}) {
	const { getFormTitle } = useForms({ forms });
	const updateChosenForm = formId => {
		const formTitle = formId ? getFormTitle(formId) : "";
		setAttributes({
			formId: formId,
			formTitle
		});
	};

	if (!formId) {
		return (
			<BlockPlaceholder>
				<ChooseForm
					label={labelText ? labelText : "Select Form"}
					formId={formId}
					forms={forms}
					onChange={updateChosenForm}
					labelText={labelText}
				/>
			</BlockPlaceholder>
		);
	}

	return (
		<React.Fragment>
			<InspectorControls>
				<ChooseForm
					formId={formId}
					forms={forms}
					onChange={updateChosenForm}
					labelText={labelText}
				/>
			</InspectorControls>
			<FormPreviewIFrame
				siteUrl={siteUrl}
				previewToken={previewToken}
				formId={formId}
			/>
		</React.Fragment>
	);
}

Edit.propTypes = {
	formId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	forms: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
	labelText: PropTypes.string,
	setAttributes: PropTypes.func.isRequired,
	siteUrl: PropTypes.string.isRequired,
	previewToken: PropTypes.string.isRequired
};
