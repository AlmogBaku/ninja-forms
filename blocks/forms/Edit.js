import React from 'react';

export default  function Edit() {
    return <React.Fragment/>
}