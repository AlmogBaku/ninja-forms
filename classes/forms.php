<?php
/**
 * Handles adding and removing forms.
 *
 * @package     Ninja Forms
 * @subpackage  Classes/Form
 * @copyright   Copyright (c) 2014, WPNINJAS
 * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
 * @since       2.9
*/

class NF_Forms {

	/**
	 * Get things started
	 * 
	 * @access public
	 * @since 2.9
	 * @return void
	 */
	public function __construct() {
		
	}

	/**
	 * Get all forms
	 * 
	 * @access public
	 * @since 2.9
	 * @return array $forms
	 */
	public function get_all( $debug = false, $show_new = false ) {
		$forms = nf_get_objects_by_type( 'form' );

		$tmp_array = array();
		foreach ( $forms as $form ) {
			$form_id = $form['id'];

			$status = Ninja_Forms()->form( $form_id )->get_setting( 'status' );
			if ( ( $status == 'new' && $show_new ) || $status != 'new' ) {
				$title = Ninja_Forms()->form( $form_id )->get_setting( 'form_title' );
				if ( strpos( $title, '_' ) === 0 ) {
					if ( $debug )
						$tmp_array[] = $form_id;
				} else {
					$tmp_array[] = $form_id;
				}
			}
		}
		return $tmp_array;
	}
}