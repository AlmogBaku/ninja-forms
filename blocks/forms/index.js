import Edit from './Edit';
import  block from './block';
import { __ } from '@wordpress/i18n';
import { registerBlockType } from '@wordpress/blocks';

registerBlockType('ninja-forms/form',{
    ...block,
    title: __('Ninja Form', 'ninja-forms'),
    edit: ({attributes,setAttributes}) => {

        return <Edit/>
    },
    save: () => null
});