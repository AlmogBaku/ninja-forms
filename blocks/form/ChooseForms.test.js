import React from "react";
import { ChooseForm } from "./Edit";
import { render, fireEvent } from "@testing-library/react";
import forms from "./forms.fixture.js";

describe("ChooseForm", () => {
	it("matches snapshot", () => {
		const { container } = render(
			<ChooseForm formId={2} forms={forms} onChange={jest.fn()} />
		);
		expect(container).toMatchSnapshot();
	});
	it("Calls on change with chosen form id", () => {
		const onChange = jest.fn();
		const labelText = "Test";
		const { getByLabelText } = render(
			<ChooseForm
				formId={2}
				forms={forms}
				onChange={onChange}
				labelText={labelText}
			/>
		);
		fireEvent.change(getByLabelText(labelText), {
			target: { value: "3" }
		});
		expect(onChange).toBeCalledTimes(1);
		expect(onChange).toBeCalledWith("3");
	});
});
