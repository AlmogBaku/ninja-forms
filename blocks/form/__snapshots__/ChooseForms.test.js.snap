// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseForm matches snapshot 1`] = `
<div>
  <div
    class="components-base-control"
  >
    <div
      class="components-base-control__field"
    >
      <label
        class="components-base-control__label"
        for="inspector-select-control-0"
      >
        Choose Form
      </label>
      <select
        class="components-select-control__input"
        id="inspector-select-control-0"
      >
        <option>
          ---
        </option>
        <option
          value="1"
        >
          Fake Form
        </option>
        <option
          value="2"
        >
          Real Form
        </option>
        <option
          value="3"
        >
          Unreal Form
        </option>
        <option
          value="4"
        >
          Mystery Form
        </option>
      </select>
    </div>
  </div>
</div>
`;
